doc/source/WorkingWithTheNodeGraph/Contexts/index.md

```diff
@@ -0,0 +1,275 @@
+
+
+## Nodes and Contexts ##
+
+In order to provide a background for why Contexts are necessary, we should look at what kind of workload a node would be up against without them.
+
+Let's consider a 3D scene. Keeping in mind [how scenes are structured](../../WorkingWithScenes/AnatomyOfAScene/index.md), imagine a scene filled with hundreds of thousands of locations. Clearly, not every location can be processed at once. Gaffer therefore needs a way to divide up the work to compute the locations one at a time.
+
+This is where Contexts come in. In Gaffer, at no point does a node operate on the entire scene/image. Contexts break the scene/image down into digestible fragments, so each node only generates the parts that they were told to, while deferring the rest. Each computing thread has its own Context, so multiple threads may be processing different parts of the scene/image in parallel, greatly improving efficiency and performance.
+
+```eval_rst
+.. figure:: images/illustrationContextsSlices.png
+    :scale: 100%
+    :alt: A scene file being sliced up by location.
+    
+    **Figure A:** Contexts allow a scene file to be processed in slices, one location at a time.
+```
+
+
+## Context flow ##
+
+While scene and image data flows down the graph, Contexts flow upstream.
+
+Computation begins when a node is queried to compute a value for its output plug, by something such as a dispatcher or an editor in the interface. At the start of the computation, whatever made the request creates a Context and sets its variables to specify the target location, set, image tile, channel, etc.
+
+The Context then passes up to any plugs that contribute to the output value. When an input plug on the queried node is connected to a plug on an upstream node, it pulls on that upstream plug to retrieve its value. If necessary, the upstream node may need to compute in turn, to produce the upstream plug's value. This pull-compute reaction occurs up through the graph until all required plug values have been calculated to create an output value for the original queried node.
+
+```eval_rst
+.. figure:: images/illustrationContextsFlow.png
+    :scale: 100%
+    :alt: Illustration of Contexts flowing through a network.
+    
+    **Figure B:** A visualization of Context and computation flow:
+```
+
+<div style="font-style: italic; font-size: 0.875em">
+<ol>
+<li>A suitable Context is created. The node’s output plug is queried. The node begins computation.</li>
+<li>The node's in plugs are computed. If connected, they pull on connected input plugs, and pass along the Context.</li>
+<li>A chain reaction of plug value pulling and Context passing ensues.</li>
+<li>Once the last input plugs in the chain are computed, input plugs begin delivering values to the plugs that pulled on them.</li>
+<li>A chain reaction of plug value returning ensues.</li>
+<li>The out plug is computed.</li>
+</ol>
+</div>
```

Comment:
Docutils doesn't support multi-paragraph captions in figures. It isn't ideal, but there's no way around this without a custom Sphinx directive.